samples-generator/lib/createFeatureMetadataExtension.ts

```diff
@@ -0,0 +1,271 @@
+        if (featureTableAttributes[attributeNames[i]] instanceof Array) {
+            humanReadable.push({
+                name: attributeNames[i],
+                value: featureTableAttributes[attributeNames[i]]
+            });
+        } else {
+            binaryReadable.push(featureTableAttributes[attributeNames[i]]);
+        }
+    }
+
+    const humanFeatureCount = assertHumanReadableFeatureTableValueArraysHaveIdenticalLength(
+        humanReadable
+    );
+    const binaryFeatureCount = assertBinaryHasIdenticalCount(binaryReadable);
+
+    if (humanFeatureCount === 0 && binaryFeatureCount === 0) {
+        throw new Error(
+            'Must have at least one binary or human readable batch table attribute'
+        );
+    }
+
+    if (
+        humanFeatureCount > 0 &&
+        binaryFeatureCount > 0 &&
+        binaryFeatureCount !== humanFeatureCount
+    ) {
+        throw new Error(
+            'Count must be identical across all batch table properties (binary or human readable)'
+        );
+    }
+
+    activeFeatureTable.featureCount = Math.max(
+        humanFeatureCount,
+        binaryFeatureCount
+    );
+
+    for (let i = 0; i < humanReadable.length; ++i) {
+        const attribute = humanReadable[i];
+        activeFeatureTable.properties[attribute.name] = {
+            values: attribute.value
+        };
+    }
+
+    if (binaryReadable.length > 0 && !defined(sharedBinaryBuffer)) {
+        throw new Error(
+            'Detected a binary attribute, but function called without binary buffer'
+        );
+    }
+
+    binaryReadable.sort(sortByByteOffset);
+    if (binaryReadable.length > 0) {
+        gltf.buffers.push({
+            byteLength: sharedBinaryBuffer.length,
+            uri:
+                'data:application/octet-stream;base64,' +
+                sharedBinaryBuffer.toString('base64')
+        });
+
+        const newBufferIndex = gltf.buffers.length - 1;
+        let bufferViewIndex = gltf.bufferViews.length;
+        let accessorIndex = gltf.accessors.length;
+
+        for (let i = 0; i < binaryReadable.length; ++i, ++accessorIndex) {
+            const batchAttribute = binaryReadable[i];
+
+            const componentType = batchAttribute.componentType;
+            const validComponentType = typeConversion.isValidWebGLDataTypeEnum(
+                componentType
+            );
+            const normalizedComponentType = validComponentType
+                ? componentType
+                : typeConversion.componentTypeStringToInteger(componentType);
+
+            const implicitBufferView = isCandidateForImplicitBufferView(
+                batchAttribute
+            );
+            if (!implicitBufferView) {
+                gltf.bufferViews.push({
+                    buffer: newBufferIndex,
+                    byteLength: batchAttribute.byteLength,
+                    byteOffset: batchAttribute.byteOffset,
+                    target: 0x8892 // ARRAY_BUFFER
```

Comment:
Maybe in the next pass, you could create an explicit Type for this too. Just to make it more readable.